polynote-frontend/polynote/ui/component/notebooklist.ts

```diff
@@ -453,6 +444,43 @@ export class BranchEl extends Disposable {
             current.expanded = false;
         }
     }
+
+    private fileHandler(evt: DragEvent) {
+        // An ugly hack to remove the highlighting effect whenever a file is dropped.
+        // This is necessary because we can't always call `evt.stopPropagation()` since it
+        // will not allow for users to automatically open and close children/nested folders on mouse drag.
+        if (evt.type === "drop") {
+            evt.stopPropagation();
+            evt.preventDefault();
+
+            // Find all parent branches (who are the grandparent node of the branch in the DOM) and remove their highlight property
+            let grandparent = this.el.parentElement?.parentElement;
+            while (grandparent?.classList.contains("highlight-branch")) {
+                grandparent.classList.remove("highlight-branch");
+                grandparent = grandparent?.parentElement?.parentElement;
+            }
+
+            // Remove the highlight property on the entire notebook list itself
+            const nbList = document.body.querySelector('.notebooks-list');
+            nbList?.classList.remove("highlight");
+        }
+
+        // handle highlighting
+        if (evt.type === "dragenter" || evt.type === "dragover") {
+            this.dragEnter = evt.target;
+            this.el.classList.add('highlight-branch');
+            this.expanded = true;
+        } else if (evt.type === "drop" || (evt.type === "dragleave" && evt.target === this.dragEnter)) {
+            this.el.classList.remove('highlight-branch');
+            this.expanded = false;
+        }
+
+        // actually handle the file
+        if (evt.type === "drop") {
+            handleFileDrop(evt, this.dispatcher, this.path);
+            this.expanded = true; // keep the folder open so the user can see where their file went
```

Comment:
Should this be the behavior on a successful file drop? Open to hearing opinions :smiley: